src/models/commented-documents.ts

```diff
@@ -0,0 +1,146 @@
+import { makeAutoObservable } from "mobx";
+import { chunk } from "lodash";
+import { Firestore } from "../lib/firestore";
+import { ClassDocument, CurriculumDocument, DocumentDocument } from "../lib/firestore-schema";
+import { getSectionTitle } from "./curriculum/section";
+import { UserModelType } from "./stores/user";
+
+export interface CurriculumDocumentInfo {
+  id: string;
+  unit: string;
+  problem: string;
+  path: string;
+  title: string;
+  numComments: number;
+}
+
+export interface StudentDocumentInfo {
+  id: string;
+  key: string;
+  title: string;
+  numComments: number;
+}
+
+export class CommentedDocumentsQuery {
+  db: Firestore;
+  user: UserModelType;
+  unit: string;
+  problem: string;
+
+  curriculumDocs: CurriculumDocumentInfo[] = [];
+  studentDocs: StudentDocumentInfo[] = [];
+
+  constructor(
+      db: Firestore,
+      user: UserModelType,
+      unit: string,
+      problem: string) {
+    makeAutoObservable(this);
+    this.db = db;
+    this.user = user;
+    this.unit = unit;
+    this.problem = problem;
+  }
+
+  getCurricumDocs(): CurriculumDocumentInfo[] {
+    return this.curriculumDocs;
+  }
+
+  getStudentDocs(): StudentDocumentInfo[] {
+    return this.studentDocs;
+  }
+
+  async queryCurriculumDocs() {
+    const cDocsRef = this.db.collection("curriculum");
+    let docsQuery;
+    if (this.user.network){
+      docsQuery = cDocsRef
+      .where("unit", "==", this.unit)
+      .where("problem", "==", this.problem)
+      .where("network","==", this.user.network);
+    } else {
+      docsQuery = cDocsRef
+      .where("unit", "==", this.unit)
+      .where("problem", "==", this.problem)
+      //for teachers not in network, look for documents matching the uid
+      .where ("uid", "==", this.user.id);
+    }
+    const result = await docsQuery.get();
+    const docs: CurriculumDocumentInfo[] = result.docs.map(doc => {
+      return {
+        id: doc.id,
+        title: "temp",
+        numComments: 0,
+        ...doc.data() as CurriculumDocument
+      };
+    });
+    const commentedDocs: CurriculumDocumentInfo[] = [];
+    const promiseArr: Promise<void>[] = [];
+    for (let doc of docs) {
+      const docCommentsRef = cDocsRef.doc(doc.id).collection("comments");
+      promiseArr.push(docCommentsRef.get().then((qs) => {
+        if (qs.empty === false) {
+          const firstCharPosition = doc.id.split("_", 4).join("_").length + 1; //first char after 4th _
+          const sectionType = doc.id.substring(firstCharPosition, doc.id.length);
+          doc = { ...doc, title: getSectionTitle(sectionType), numComments: qs.size };
+          commentedDocs.push(doc);
+        }
+      }));
+    }
+    await Promise.all(promiseArr);
+    this.curriculumDocs = commentedDocs;
+  }
+
+  async queryStudentDocs() {
```

Comment:
This will including teacher documents too. So maybe the name `getUserDocs` would be better.